networking_ccloud/services/trunk/driver.py

```diff
@@ -0,0 +1,227 @@
+
+    def subport_create(self, resource, event, trunk_plugin, payload):
+        trunk_port = self._get_parent_port(payload.context, payload.states[0].port_id)
+        if not trunk_port:
+            return
+        self._bind_subports(payload.context, trunk_port, payload.states[0], payload.metadata['subports'])
+
+    def subport_delete(self, resource, event, trunk_plugin, payload):
+        trunk_port = self._get_parent_port(payload.context, payload.states[0].port_id)
+        if not trunk_port:
+            return
+        self._unbind_subports(payload.context, trunk_port, payload.states[0], payload.metadata['subports'])
+
+    def _bind_subports(self, context, trunk_port, trunk, subports):
+        for subport in subports:
+            LOG.info("Adding subport %s trunk port %s of trunk %s", subport.port_id, trunk.port_id, trunk.id)
+            binding_profile = trunk_port.get(pb_api.PROFILE)
+
+            # note, that this information is only informational
+            binding_profile[cc_const.TRUNK_PROFILE] = {
+                'segmentation_type': subport.segmentation_type,
+                'segmentation_id': subport.segmentation_id,
+            }
+
+            port_data = {
+                p_api.RESOURCE_NAME: {
+                    pb_api.HOST_ID: trunk_port.get(pb_api.HOST_ID),
+                    pb_api.VNIC_TYPE: trunk_port.get(pb_api.VNIC_TYPE),
+                    pb_api.PROFILE: binding_profile,
+                    'device_owner': trunk_const.TRUNK_SUBPORT_OWNER,
+                    'device_id': trunk_port.get('device_id'),
+                }
+            }
+            self.core_plugin.update_port(context, subport.port_id, port_data)
+        if len(subports) > 0:
+            trunk.update(status=trunk_const.TRUNK_ACTIVE_STATUS)
+
+    def _unbind_subports(self, context, trunk_port, trunk, subports):
+        for subport in subports:
+            LOG.info("Adding subport %s trunk port %s of trunk %s", subport.port_id, trunk.port_id, trunk.id)
+            binding_profile = trunk_port.get(pb_api.PROFILE)
+
+            # note, that this is only informational
+            if cc_const.TRUNK_PROFILE in binding_profile:
+                del binding_profile[cc_const.TRUNK_PROFILE]
+
+            port_data = {
+                p_api.RESOURCE_NAME: {
+                    pb_api.HOST_ID: None,
+                    pb_api.VNIC_TYPE: None,
+                    pb_api.PROFILE: binding_profile,
+                    'device_owner': '',
+                    'device_id': '',
+                    'status': nl_const.PORT_STATUS_DOWN,
+                },
+            }
+            self.core_plugin.update_port(context, subport.port_id, port_data)
+
+            if len(trunk.sub_ports) - len(subports) > 0:
+                trunk.update(status=trunk_const.TRUNK_ACTIVE_STATUS)
+            else:
+                LOG.info("Last subport was removed from trunk %s, setting it to state DOWN", trunk.id)
```

Comment:
You do not set to state down here. Does that happen automatically and this is just an FYI?

Should this rather be `trunk_const.TRUNK_DOWN_STATUS` or something therelike?

Does updating the trunk still work if we run this code in `AFTER_DELETE` for the trunk?

Reply:
After an operation of the trunk it is automatically set to down if we don't do anything against that.